src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -21,29 +24,42 @@ function HeatmapVis(props: Props): JSX.Element {
   const values = dataArray.data as number[];
 
   const {
-    dataDomain,
     customDomain,
     scaleType,
     colorMap,
     keepAspectRatio,
     showGrid,
-    initDataDomain,
+    resetDomains,
   } = useHeatmapConfig();
 
-  // width / height <=> cols / rows
-  const aspectRatio = keepAspectRatio ? cols / rows : undefined;
+  const dataDomain = useMemo(() => findDomain(values), [values]);
```

Comment:
The data domain is now computed here instead of in the config. This means that it is never stale during the render, even when slicing through dimensions.